packages/iov-rise/src/riseconnection.spec.ts

```diff
@@ -182,33 +182,91 @@ describe("RiseConnection", () => {
+      }
+
+      // by pubkey
+      {
+        const query: BcpPubkeyQuery = {
+          pubkey: {
+            algo: Algorithm.Ed25519,
+            data: fromHex(
+              "ac681190391fe048d133a60e9b49f7ac0a8b0500b58a9f176b88aee1e79fe735",
+            ) as PublicKeyBytes,
+          },
+        };
+        const nonce = await connection.getNonce(query);
+        // nonce is current unix timestamp +/- one second
+        expect(nonce.toNumber()).toBeGreaterThanOrEqual(Date.now() / 1000 - 1);
+        expect(nonce.toNumber()).toBeLessThanOrEqual(Date.now() / 1000 + 1);
+      }
+
+      connection.disconnect();
+    });
+  });
 
-    // by address
-    {
-      const query: BcpAddressQuery = { address: "5399275477602875017R" as Address };
-      const nonce = await connection.getNonce(query);
-      // nonce is current unix timestamp +/- one second
-      expect(nonce.toNumber()).toBeGreaterThanOrEqual(Date.now() / 1000 - 1);
-      expect(nonce.toNumber()).toBeLessThanOrEqual(Date.now() / 1000 + 1);
-    }
-
-    // by pubkey
-    {
-      const query: BcpPubkeyQuery = {
+  describe("getNonces", () => {
+    it("can get 0/1/2 nonces", async () => {
+      const connection = await RiseConnection.establish(base);
+
+      const addressQuery: BcpAddressQuery = { address: "5399275477602875017R" as Address };
+      const pubkeyQuery: BcpPubkeyQuery = {
         pubkey: {
           algo: Algorithm.Ed25519,
           data: fromHex("ac681190391fe048d133a60e9b49f7ac0a8b0500b58a9f176b88aee1e79fe735") as PublicKeyBytes,
         },
       };
-      const nonce = await connection.getNonce(query);
-      // nonce is current unix timestamp +/- one second
-      expect(nonce.toNumber()).toBeGreaterThanOrEqual(Date.now() / 1000 - 1);
-      expect(nonce.toNumber()).toBeLessThanOrEqual(Date.now() / 1000 + 1);
-    }
 
-    connection.disconnect();
+      // by address, 0 nonces
+      {
+        const nonces = await connection.getNonces(addressQuery, 0);
+        expect(nonces.length).toEqual(0);
+      }
+
+      // by address, 1 nonces
+      {
+        const nonces = await connection.getNonces(addressQuery, 1);
+        expect(nonces.length).toEqual(1);
+      }
+
+      // by address, 2 nonces
+      {
+        const nonces = await connection.getNonces(addressQuery, 2);
+        expect(nonces.length).toEqual(2);
+        expect(nonces[1].toNumber()).toEqual(nonces[0].toNumber());
+      }
+
+      // by pubkey, 0 nonces
+      {
+        const nonces = await connection.getNonces(pubkeyQuery, 0);
+        expect(nonces.length).toEqual(0);
+      }
+
+      // by pubkey, 1 nonces
+      {
+        const nonces = await connection.getNonces(pubkeyQuery, 1);
+        expect(nonces.length).toEqual(1);
+      }
+
+      // by pubkey, 2 nonces
+      {
+        const nonces = await connection.getNonces(pubkeyQuery, 2);
+        expect(nonces.length).toEqual(2);
+        expect(nonces[1].toNumber()).toEqual(nonces[0].toNumber());
```

Comment:
Yup, same number.
If I want to send the same person the same amount in two separate transactions in the same block, this will fail.

Maybe super edge case, I still like a minor change on each returned nonce

Reply:
done, thanks